pkg/scheduler/framework/plugins/noderesources/fit.go

```diff
@@ -226,6 +232,32 @@ func computePodResourceRequest(pod *v1.Pod, opts ResourceRequestsOptions) *preFi
 	return result
 }
 
+// withDeviceClass adds resource to device class mapping to preFilterState.
+func withDeviceClass(result *preFilterState, draManager framework.SharedDRAManager) *fwk.Status {
+	hasExtendedResource := false
+	for rName, rQuant := range result.ScalarResources {
+		// Skip in case request quantity is zero
+		if rQuant == 0 {
+			continue
+		}
+
+		if v1helper.IsExtendedResourceName(rName) {
+			hasExtendedResource = true
+			break
+		}
+	}
+	if hasExtendedResource {
+		resourceToDeviceClass, err := extended.DeviceClassMapping(draManager)
+		if err != nil {
+			return fwk.AsStatus(err)
+		}
+		if len(resourceToDeviceClass) != 0 {
+			result.resourceToDeviceClass = resourceToDeviceClass
+		}
```

Comment:
One important thing:
We should always write the `resourceToDeviceClass` here, even if it's empty, because we distinguish nil map from empty map.
```suggestion
		result.resourceToDeviceClass = resourceToDeviceClass
		if result.resourceToDeviceClass == nil {
			result.resourceToDeviceClass = make(...)
		}
```